fix bug with make_array/array_wrapper introduced in Boost v1.61; cfr.:
* https://svn.boost.org/trac/boost/ticket/12516
* https://github.com/boostorg/serialization/pull/46 (source of this patch)
--- boost_1_63_0/boost/serialization/array.hpp.orig	2017-02-23 10:38:00.487842679 +0100
+++ boost_1_63_0/boost/serialization/array.hpp	2017-02-23 10:38:13.918969000 +0100
@@ -44,7 +44,7 @@
     // note: I would like to make the copy constructor private but this breaks
     // make_array.  So I try to make make_array a friend - but that doesn't
     // build.  Need a C++ guru to explain this!
-    template<class S>
+    template<class T, class S>
     friend const boost::serialization::array_wrapper<T> make_array( T* t, S s);
 
     array_wrapper(const array_wrapper & rhs) :
